### PR TITLE
Advanced upload page

### PR DESCRIPTION
**Pull request by Seneli**
## This PR addresses the following milestones/tasks:
- 5.3 #3 

## Things not included in this PR:
- Logos are not added in since the code for adding in font-awesome is in the dashboard PR.
- This PR does everything except send the final request to the back end, but the function that will do that has been identified with a comment. 
- If we have time (unlikely) we could animate the components in this PR but it's not necessary. 

## Summary of changes file by file:

- `Upload.tsx` - the component that encapsulates everything shown in the upload page. It is the most high level component in this PR.

- `CarouselController.tsx` is the left hand side of the component thingy shown below. `CarouselBody.tsx` is the right side of the component thingy shown below.
Screen Shot 2023-01-23 at 3 16 53 PM
 
- `UploadStep.tsx` - the entirety of one of the upload steps. It's a wrapper for the radio dropdowns or file upload component listed right after this :) 
Screen Shot 2023-01-23 at 3 16 11 PM

- `RadioDropdown.tsx` - has a list of dropdown options that you can choose from
Screen Shot 2023-01-23 at 3 15 47 PM

- `FileUpload.tsx` - component that handles the upload of the file. 
Screen Shot 2023-01-23 at 3 15 33 PM

### CODE REVIEW

**Comment by z-arnott** [comment]
can you remove "input manually" 
I don't see a text box entry yet - and we decided as a group during design that this adds complexity without much benefit

and remove step 2 completely

**Comment by z-arnott** [comment]
change 'submit query' to 'submit' in Step 3, pictured below
image

**Comment by z-arnott** [comment]
what I don't see in the UI or in the code (I might have missed it) is a way to determine the filetype 
it would be simplest to have the user specify

File: client/src/components/FileUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+interface FileUploadProps {
+  format: string | undefined;
+  uploadMethod: string | undefined;
+}
+
+const FileUpload = ({ format, uploadMethod }: FileUploadProps) => {
+  const [file, setFile] = useState();
+  const [uploadMessage, setUploadMessage] = useState('');
+
+  const setUploadStateMessage = () => {
```
Comment (z-arnott) [inline client/src/components/FileUpload.tsx]: how are you determining filetype?

**Review by z-arnott — CHANGES_REQUESTED** [review]

**Comment by Seneli** [comment]
I need some help with this one. When I console.log the request I thought the query params would at least send, but that and the files don't make it to the back end. I'm not sure why but if someone else can debug this issue for now it would help me move on :") 

Screen Shot 2023-01-31 at 7 43 24 AM

Screen Shot 2023-01-31 at 7 44 16 AM

**Comment by z-arnott** [comment]
I was able to  send requests with the following hardcoded values  in FIleUpload.tsx
image

**Comment by Seneli** [comment]
This is the error I am getting now. I think I am able to send things well but I couldn't figure out how to resolve this. 

Screen Shot 2023-02-04 at 3 00 40 PM

**Comment by Seneli** [comment]
I think this is going to be my final commit to the upload page so here's my changes since we talked last time :) 

#### 1. Added a function to upload the file to the back end. 
- There's some kinda vague error reporting  to this and file type validation. You can test it out to see more of what that looks like.
- Currently, input validation is done through getting the file type extension via regex. It's not perfect but it's something.
Screen Shot 2023-02-04 at 6 36 36 PM
Screen Shot 2023-02-04 at 6 37 09 PM



#### 2. Added option to select that you wanted to upload a SPDX document in the tag value.
Screen Shot 2023-02-04 at 6 35 11 PM
- This also included making it so whenever a user changed their file format they had to re-specify their file type, since the options between the 2 aren't the same. 

#### 3. Added icons to the left side
Screen Shot 2023-02-04 at 6 17 31 PM

**Review by z-arnott — APPROVED** [review]
Looks great - thanks for making those changes
Tested it with back end integration today and it's working great :)

**Pull request merged**